.github/workflows/build-on-macos.yml

```diff
@@ -0,0 +1,148 @@
+name: Build on macOS 15 1
+
+on:
+  push:
+    branches: [ "dev", "candidate", "release", "jshooks" ]
+  pull_request:
+    branches: [ "dev", "candidate", "release", "jshooks" ]
+
+env:
+  XAHAUD_DEPS: /Users/runner/xahaud_deps
+
+# This means that if you push a new commit to a branch or PR that previously running
+# jobs are canceled and re-run. This is useful for preventing duplicated work.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-macos:
+    runs-on: macos-15
```

Comment:
You could use the self hosted runner if you want. See the conan branch for the example.